client.go

```diff
@@ -692,16 +693,26 @@ func newLink(s *Session, r *Receiver, opts []LinkOption) (*link, error) {
 	if isReceiver {
 		attach.Role = roleReceiver
 		attach.Source = &source{
-			Address: l.address,
+			Address: l.sourceAddress,
 			Dynamic: l.dynamicAddr,
 			Filter:  l.filters,
 		}
+		if l.targetAddress != "" {
```

Comment:
Doesn't look pretty, probably I can rewrite it into something like this in case it's necessary to configure all source/target attributes:
```go
func LinkSourceAddress(addr string) LinkOption {
	return func(l *link) error {
		if l.source == nil {
			l.source = &source{}
		}
		l.source.Address = addr
		return nil
	}
}
```

Reply:
Yeah, I like that approach.

If you do go that route, please use `new(source)` rather than `&source{}`. There's no effective difference but I personally prefer to use `new()` when I'm not initializing any fields and would like to keep that consistent.

Reply:
Done. Also moved `l.filters` to `l.source`, not sure what we should do with `l.dynamicAddr` leave it as it's now or split it into two different funcs.